pkg/cluster/cache.go

```diff
@@ -46,23 +46,36 @@ func (c *kubeFedClusterClients) getFedCluster(name string) (*FedCluster, bool) {
 	return cluster, ok
 }
 
-func (c *kubeFedClusterClients) getFirstFedCluster() (*FedCluster, bool) {
+func (c *kubeFedClusterClients) getFedClustersByType(clusterType Type) []*FedCluster {
 	c.RLock()
 	defer c.RUnlock()
+	clusters := make([]*FedCluster, 0, len(c.clusters))
 	for _, cluster := range c.clusters {
-		return cluster, true
+		if cluster.Type == clusterType {
+			clusters = append(clusters, cluster)
+		}
 	}
-	return nil, false
+	return clusters
 }
 
 // GetFedCluster returns a kube client for the cluster (with the given name) and info if the client exists
 func GetFedCluster(name string) (*FedCluster, bool) {
 	return clusterCache.getFedCluster(name)
 }
 
-// GetFirstFedCluster returns a first kube client from the cache of clusters and info if such a client exists
-func GetFirstFedCluster() (*FedCluster, bool) {
-	return clusterCache.getFirstFedCluster()
+// GetHostCluster returns the kube client for the host cluster from the cache of the clusters
+// and info if such a client exists
+func GetHostCluster() (*FedCluster, bool) {
+	clusters := clusterCache.getFedClustersByType(Host)
+	if len(clusters) == 0 {
+		return nil, false
+	}
+	return clusters[0], true
+}
+
+// GetMemberClusters returns the kube clients for the host clusters from the cache of the clusters
+func GetMemberClusters() []*FedCluster {
```

Comment:
Please change the function so it takes a name of the required member cluster and returns the single cluster:
```go
func GetFedCluster(name string) (*FedCluster, bool)
```

Reply:
...so it is easier to use in host operator - for example:
```go
fedCluster, ok := cluster.GetMemberCluster(account.TargetCluster)
if !ok {
	return fmt.Errorf("the fedCluster %s not found in the registry", account.TargetCluster)
}
```

Reply:
We need a function to get all the available member clusters when we approving/provisioning the user from host. It's for the logic which will check all the cluster's capacity and decide which one to choose to provision the user to.

For the case you mention we can still use the existing `GetFedCluster(name)`, can't we?

Reply:
yeah, I thought that you replaced the function - sorry for confusion... :-)